src/ppx/graphics_util.cpp

```diff
@@ -1510,5 +1706,65 @@ Result CreateMeshFromFile(
     return ppx::SUCCESS;
 }
 
+// -------------------------------------------------------------------------------------------------
+
+Result LoadFramesFromRawVideo(
+    const std::filesystem::path&    path,
+    grfx::Format                    format,
+    uint32_t                        width,
+    uint32_t                        height,
+    std::vector<std::vector<char>>* pFrames)
+{
+    PPX_ASSERT_NULL_ARG(pFrames);
+
+    const grfx::FormatDesc* formatDesc = grfx::GetFormatDescription(format);
+    if (formatDesc == nullptr) {
+        PPX_LOG_ERROR("Failed to fetch information for texture format " << format);
+        return ppx::ERROR_FAILED;
+    }
+
+    uint32_t frameSize = 0; // As measured in bytes, not pixels.
+    if (formatDesc->isPlanar) {
+        std::optional<grfx::FormatPlaneDesc> formatPlanes = grfx::GetFormatPlaneDescription(format);
+        PPX_ASSERT_MSG(formatPlanes.has_value(), "No planes found for format " << format);
+        frameSize = GetPlanarImageSizeInBytes(*formatDesc, *formatPlanes, width, height);
+    }
+    else {
+        frameSize = formatDesc->bytesPerTexel * width * height;
+    }
+
+    ppx::fs::File file;
+    if (!file.Open(path)) {
+        PPX_ASSERT_MSG(false, "Cannot open the file!");
+        return ppx::ERROR_FAILED;
+    }
+    const size_t fileSize = file.GetLength();
+
+    std::vector<char> buffer(frameSize);
+    size_t            totalRead = 0;
+    while (totalRead < fileSize) {
+        const size_t bytesRead = file.Read(buffer.data(), frameSize);
```

Comment:
Is this safe to do after a std::move operation? Would buffer have frameSize capacity?